File: programmers/탑.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> heights) {
    vector<int> answer;
    answer.push_back(0);
    for(int i=1; i<heights.size();i++){
        bool ZeroCheck=true;
        for(int j=i; j>=0; j--){
            if(heights[i]<heights[j]){
                answer.push_back(j+1);
                ZeroCheck=false;
                break;
            }
        }
        if(ZeroCheck)
             answer.push_back(0);
            
    }
    return answer;
}

File: programmers/탑.java
class Solution {
    public int[] solution(int[] heights) {

        int[] answer = new int[heights.length];
        //오른쪽 부터 시작 
        for(int i=heights.length-1; i>=1; i--){
            for(int j=i-1; j>=0; j--){
                // 자신보다 높은 건물이 있으면 탈출
                if(heights[i]<heights[j]){
                    answer[i]=j+1;
                    break;
                }
            }
        }
        return answer;
    }
}